aten/src/THC/THCGenerator.h

```diff
@@ -0,0 +1,19 @@
+#ifndef THC_GENERATOR_INC
+#define THC_GENERATOR_INC
+
+#include <mutex>
+
+typedef struct THCGeneratorState {
+  struct curandStateMtgp32* gen_states;
+  struct mtgp32_kernel_params *kernel_params;
+  int initf;
+  uint64_t initial_seed;
+  int64_t philox_seed_offset;
+} THCGeneratorState;
+
+struct THCGenerator {
+  std::mutex mutex; /* mutex for using this generator */
+  THCGeneratorState gen_state;
```

Comment:
nit: can we call this `state`? The generator object will usually be called something like `gen` or `generator` and `gen->gen_state` is quite verbose